airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java

```diff
@@ -196,7 +199,7 @@ public ConfigurationApi(final ConfigRepository configRepository,
         workerEnvironment,
         logConfigs,
         eventRunner);
-
+    stateHandler = new StateHandler(new StatePersistence(configsDatabase));
```

Comment:
Nit:  Consider constructing the `StatePersistence` object outside of this class and passing it in as an argument.  It will make the transition to dependency injection easier.

Reply:
legit. done.